rfcs/text/0008-threat-intel.md

```diff
@@ -75,7 +81,13 @@ Stage 3: Add or update all remaining field definitions. The list should now be e
 Stage 1: Describe at a high-level how these field changes will be used in practice. Real world examples are encouraged. The goal here is to understand how people would leverage these fields to gain insights or solve problems. ~1-3 paragraphs.
 -->
 
-The additions to threat.* fields and nested use will be used to represent data collected threat intelligence sources. A new rule type Indicator match will be introduced in 7.10 and ECS threat fields will enable a new category of detection alerts that matches incoming log and event data against threat intelligence sources and generates an alert. Additionally in the future we will be able to enrich events and alerts with context from threat intelligence to assist analysts in their investigative workflows.
+The additions described above will be used to enable cyber threat intelligence capabilities in Elastic Security solution. A new rule type Indicator match will be introduced in 7.10 and the proposed ECS updates will enable a new category of detection alerts that match incoming log and event data against threat intelligence sources. Additionally in the future we will also develop enrichment flows that add context from threat intelligence to alerts and events to assist analysts in their investigative workflows.
+
+There are two primary uses for these fields.
+
+1. *Storing threat intelligence as an event document in threat index(s).* Threat intelligence data will be collected from multiple sources stored in threat indices. The ECS fields proposed here will be used to structure the documents collected from various sources. 
+
+2. *Adding threat intelligence match/enrichment to another document which could be in a source event index or signals index.* The Indicator Match Rule will be used to generate signals when a match occurs between a source event and threat intelligence document. The ECS fields proposed here will be used to add the enrichment and threat intel context in the signal document. 
 
 ## Source data
```

Comment:
Adding this for context, but this may not be the best place.

```suggestion
### Proposed enrichment pipeline mechanics pseudocode

1. Original document completes its standard pipeline for the given source (i.e. filebeat module pipeline)
2. Original document is sent to "threat lookup" pipeline
3. For each indicator type, we perform the following (a file sha256 for example):
    - if exists "file.hash.sha256":
        - enrich processor:
            "policy_name": "file-sha256-policy",
            "field" : "file.hash.sha256",
            "target_field": "threat_match",
            "max_matches": "1"
        - policy file-sha256-policy:
            "match": {
                "indices": "threat-*",
                "match_field": "file.hash.sha256",
                "enrich_fields": ["event", "file", "ioc", "malware", "threat_actor"]
            }
    - rename:
        field: "threat_match.file"
        target: "threat_match.ioc.file"
    - set:
        field: "theat_match.matched"
        value: "sha256"
    - append:
        field: "threat"
        value: "{{ threat_match }}"
    - remove:
        field: "threat_match"

**NOTE**: There may be some optimization on which enrichments we attempt based upon the event categorization fields. For instance, we know that data that presents the netflow model or "interface" doesn't contain a sha256 hash. Since those categorization fields are lists, if data presented as both netflow and file (for whatever reason), then we'd check both network-related lookups and file-related lookups

## Source data
```